sparts/tasks/dbus.py

```diff
@@ -134,9 +143,21 @@ def initTask(self):
     def mainloop(self):
         return self.mainloop_task.mainloop
 
+    def asyncRun(self, cb, *args):
```

Comment:
I'm thinking something like this might be cleaner / more easier to reuse:

```
def asyncRun(self, cb, *args):
    def _future_execute(f, cb, *args, **kwargs):
        try:
            # Only execute `cb` if the future wasn't cancelled
            if f.set_running_or_notify_cancel():
                f.set_result(cb(*args, **kwargs))
        except Exception as e:
            f.set_exception(e)

    def _future_cancel(f, handle):
        if f.cancelled():
            glib.source_remove(handle)

    f = Future()
    handle = glib.idle_add(partial(_future_execute, f, cb, *args, **kwargs))
    f.add_done_callback(partial(_future_cancel, f, handle))
    return f
```

Then the caller can choose to call `f.result()` on their own,  with or without a timeout as desired, without having to implement future semantics in the callbacks or wrapping cb with the try:/except: blocks.  It support callers cancelling the idle-added function with native glib apis to remove the event source id as appropriate as well.

Thoughts?